test/test_common/environment.cc

```diff
@@ -65,12 +68,17 @@ bool TestEnvironment::shouldRunTestForIpVersion(const Network::Address::IpVersio
 
 std::vector<Network::Address::IpVersion> TestEnvironment::getIpVersionsForTest() {
   std::vector<Network::Address::IpVersion> parameters;
-  if (TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v4)) {
-    parameters.push_back(Network::Address::IpVersion::v4);
-  }
-
-  if (TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v6)) {
-    parameters.push_back(Network::Address::IpVersion::v6);
+  for (auto version : {Network::Address::IpVersion::v4, Network::Address::IpVersion::v6}) {
+    if (TestEnvironment::shouldRunTestForIpVersion(version)) {
+      parameters.push_back(version);
+      if (!Network::Test::supportsIpVersion(version)) {
+        Logger::Registry::getLog(Logger::Id::testing)
+            .warn(
```

Comment:
Is there any reason to not just outright fail/exit at this point? I don't mind either way but curious on the rationale.